pkg/gits/git_cli.go

```diff
@@ -660,7 +660,12 @@ func (g *GitCLI) GetPreviousGitTagSHA(dir string) (string, error) {
 	if err != nil {
 		return "", fmt.Errorf("failed to find previous tag for project in %s : %s", dir, err)
 	}
+
+	if len(previousTag) != 40 {
```

Comment:
I am confused. In which cases would _previousTag_ ever be a sha. IMO the whole function is broken.
AFAIK, `git describe --abbrev=0 --debug --tags --always` will always only display tag names, no shas. From the [docs](https://git-scm.com/docs/git-describe):

> The command finds the most recent tag that is reachable from a commit. If the tag points to the commit, then only the tag is shown.

Only if it is not a tag or the `--long` option is specified you get a human-readable object name of the form "v0.0.1-1-g23fece4" where the part after the 'g' is the 7 char abbreviation of the sha.

So either one needs to use the `--long` options and extract the abbreviated sha using a regexp or similar or one needs to always determine the actual sha for `previousTag` given that it is a tag name. Something equivalent to:

```
$ git rev-list < previousTag>
```

Also for the case where there is no previous tag I would think this function throws an error. Is this what we want?